infrastructure/network/addressmanager/addressmanager.go

```diff
@@ -29,9 +28,20 @@ type addressKey struct {
 	address ipv6
 }
 
+// Levels of addresses
+const (
```

Comment:
These levels are somewhat confusing.

I'd rather: 
1. Have `const maximumFailedConnectionsBeforeRemoving = 3`
2. Keep `connectionFailedCount`
3. On successful connection: 
3.1.  connectionFailedCount = 0
4. On failed connection:
4.1. connectionFailedCount++
4.2. if  connectionFailedCount >= maximumFailedConnectionsBeforeRemoving: remove

Reply:
You are right. Will do it on my big PR.